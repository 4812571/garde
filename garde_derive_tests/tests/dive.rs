#[path = "./util/mod.rs"]
mod util;

#[derive(Debug, garde::Validate)]
struct Inner<'a> {
    #[garde(length(min = 1))]
    field: &'a str,
}

#[derive(Debug, garde::Validate)]
struct Test<'a> {
    #[garde(dive)]
    field: Inner<'a>,
}

#[test]
fn email_valid() {
    util::check_ok(
        &[Test {
            field: Inner { field: "asdf" },
        }],
        &(),
    )
}

#[test]
fn email_invalid() {
    util::check_fail!(
        &[Test {
            field: Inner { field: "" }
        }],
        &()
    )
}
