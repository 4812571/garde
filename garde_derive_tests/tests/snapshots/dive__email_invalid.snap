---
source: garde_derive_tests/tests/dive.rs
expression: snapshot
---
Test {
    field: Inner {
        field: "",
    },
}
value.field.field: length is lower than 1


